packages/vue-directive/src/clickoutside.ts

```
@@ -35,16 +35,18 @@ if (!isServer) {
-      el.contains(mouseup.target) ||
-      el.contains(mousedown.target) ||
-      el === mouseup.target ||
-      (popperElm && (popperElm.contains(mouseup.target) || popperElm.contains(mousedown.target)))
-    ) {
+    // 使用 event.composedPath() 来处理 Shadow DOM 场景。
+    // composedPath() 会返回事件的完整路径，即使事件穿过了 Shadow DOM 的边界。
+    // 这确保了即使 popperElm 在 Shadow DOM 外部（或组件本身在 Shadow DOM 内部），
+    // 我们也能准确判断点击是否发生在组件或其 popper 内部。
+    const mousedownPath = (mousedown?.composedPath && mousedown.composedPath()) || [mousedown?.target]
+    const mouseupPath = (mouseup?.composedPath && mouseup.composedPath()) || [mouseup.target]
+    const isClickInEl = mousedownPath.includes(el) || mouseupPath.includes(el)
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)